core/src/decoder.rs

```diff
@@ -823,6 +815,49 @@ impl Decoder {
 	}
 }
 
+/// Kept around for backwards-compatibility with old address struct
```

Comment:
Is "old" < 1042?

Reply:
Maybe move `need_more_than` inside `decode_old_address()` to keep them bundled together?

Reply:
> Is "old" < 1042?

Anything before v0.8.26 or 26/2026/46 on polkadot/kusama/westend respectively I think